File: modules/admin-modules-installer/src/test/js/configuration.js
/**
 * Copyright (c) Codice Foundation
 *
 * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
 * version 3 of the License, or any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
 * <http://www.gnu.org/licenses/lgpl.html>.
 *
 **/
/*jshint strict:false*/
/*global CasperError, console, phantom, require, casper*/
//var x = require('casper').selectXPath;
casper.options.viewportSize = {width: 2452, height: 868};
casper.test.begin('Configuration View test', function(test) {
    casper.start('http://localhost:8383');

    casper.waitForSelector('#welcome',
    function success() {
        test.pass('Found welcome container div');
    },
    function fail() {
        test.fail('Did not find welcome container div');
    });

    casper.waitForSelector('#configuration',
    function success() {
        test.pass('Found configuration container div');
    },
    function fail() {
        test.fail('Did not find configuration container div');
    });

    casper.waitFor(function() {
       return this.evaluate(function() {
           return document.querySelectorAll('#startStep').length === 1;
       });
    }, function() {
       test.assertExists('#startStep');
       this.click('#startStep');
    }, function() {
       test.fail('Failed start button');
    });

    casper.waitFor(function() {
      return this.evaluate(function() {
          return document.querySelectorAll('#config-form').length === 1;
      });
    }, function() {
      test.pass('Found config form');
    }, function() {
      test.fail('Failed finding config form');
    });

    casper.then(function () {
        this.fill('form#config-form', { host: "" }, false);
    });

    casper.waitFor(function() {
       return this.evaluate(function() {
           return document.querySelectorAll('#nextStep').length === 1;
       });
    }, function() {
       test.assertExists('#nextStep');
       this.click('#nextStep');
    }, function() {
       test.fail('Failed next button');
    });

    casper.waitFor(function() {
      return this.evaluate(function() {
          return document.querySelectorAll('#config-form').length === 1;
      });
    }, function() {
      test.pass('Could not submit invalid hostname');
    }, function() {
      test.fail('Submitted valid hostname');
    });

    casper.then(function () {
        this.fill('form#config-form', { host: "localhost" }, false);
    });

    casper.then(function () {
        this.fill('form#config-form', { port: "blah" }, false);
    });

    casper.waitFor(function() {
       return this.evaluate(function() {
           return document.querySelectorAll('#nextStep').length === 1;
       });
    }, function() {
       test.assertExists('#nextStep');
       this.click('#nextStep');
    }, function() {
       test.fail('Failed next button');
    });

    casper.waitFor(function() {
      return this.evaluate(function() {
          return document.querySelectorAll('#config-form').length === 1;
      });
    }, function() {
      test.pass('Could not submit incorrect port value');
    }, function() {
      test.fail('Submitted incorrect port value');
    });

    casper.then(function () {
        this.fill('form#config-form', { port: "" }, false);
    });

    casper.waitFor(function() {
       return this.evaluate(function() {
           return document.querySelectorAll('#nextStep').length === 1;
       });
    }, function() {
       test.assertExists('#nextStep');
       this.click('#nextStep');
    }, function() {
       test.fail('Failed next button');
    });

    casper.waitFor(function() {
      return this.evaluate(function() {
          return document.querySelectorAll('#config-form').length === 1;
      });
    }, function() {
      test.pass('Could not submit incorrect port value');
    }, function() {
      test.fail('Submitted incorrect port value');
    });

    casper.then(function () {
        this.fill('form#config-form', { port: "8181" }, false);
    });

    casper.waitFor(function() {
       return this.evaluate(function() {
           return document.querySelectorAll('#nextStep').length === 1;
       });
    }, function() {
       test.assertExists('#nextStep');
       this.click('#nextStep');
    }, function() {
       test.fail('Failed next button');
    });

    casper.waitFor(function() {
      return this.evaluate(function() {
          return document.querySelectorAll('#config-form').length === 0;
      });
    }, function() {

      test.pass('Submitted correct form value and continued');
    }, function() {
      test.fail('Submitted correct form value but did not continue');
    });

   casper.run(function() {test.done();});
});

File: handler/security-handler-anonymous/src/main/java/org/codice/ddf/security/handler/anonymous/configuration/Configuration.java
/**
 * Copyright (c) Codice Foundation
 *
 * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
 * General Public License as published by the Free Software Foundation, either version 3 of the
 * License, or any later version.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
 * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
 * is distributed along with this program and can be found at
 * <http://www.gnu.org/licenses/lgpl.html>.
 *
 **/
package org.codice.ddf.security.handler.anonymous.configuration;

import net.minidev.json.JSONObject;
import net.minidev.json.JSONValue;
import org.apache.commons.lang.StringUtils;
import org.apache.felix.webconsole.BrandingPlugin;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.activation.MimeType;
import javax.activation.MimeTypeParseException;
import javax.servlet.http.HttpServletRequest;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.core.Context;
import javax.ws.rs.core.Response;
import javax.ws.rs.core.UriInfo;
import java.io.ByteArrayInputStream;

/**
 * Stores external configuration properties.
 *
 * @author ddf.isgs@lmco.com
 */
@Path("/")
public class Configuration {
    private static final Logger LOGGER = LoggerFactory.getLogger(Configuration.class);

    private static Configuration uniqueInstance;

    private String header = "";

    private String footer = "";

    private String style = "";

    private String textColor = "";

    private BrandingPlugin branding;

    private static MimeType mimeType = null;

    private Configuration() {
    }

    @GET
    @Path("/config")
    public Response getDocument(@Context UriInfo uriInfo, @Context HttpServletRequest httpRequest) {
        Response response;
        JSONObject configObj = new JSONObject();
        configObj.put("text", header);
        configObj.put("footer", footer);
        configObj.put("style", style);
        configObj.put("textColor", textColor);
        configObj.put("branding", getProductName());

        String configString = JSONValue.toJSONString(configObj);
        response = Response.ok(new ByteArrayInputStream(configString.getBytes()), mimeType.toString()).build();

        return response;
    }

    /**
     * @return a unique instance of {@link Configuration}
     */
    public static synchronized Configuration getInstance() {

        if (uniqueInstance == null) {
            uniqueInstance = new Configuration();
        }

        return uniqueInstance;
    }

    public String getHeader() {
        return header;
    }

    public void setHeader(String header) {
        this.header = header;
    }

    public String getFooter() {
        return footer;
    }

    public void setFooter(String footer) {
        this.footer = footer;
    }

    public String getStyle() {
        return style;
    }

    public void setStyle(String style) {
        this.style = style;
    }

    public String getTextColor() {
        return textColor;
    }

    public void setTextColor(String textColor) {
        this.textColor = textColor;
    }

    public String getProductName() {
        if (branding != null) {
            // Remove the version number
            return StringUtils.substringBeforeLast(branding.getProductName(), " ");
        } else {
            return "";
        }
    }

    public BrandingPlugin getBranding() {
        return branding;
    }

    public void setBranding(BrandingPlugin branding) {
        this.branding = branding;
    }

    public void setMimeType(String mimeType) {
        MimeType mime = null;

        try {
            mime = new MimeType(mimeType);
        } catch (MimeTypeParseException e) {
            LOGGER.warn("Failed to create mimetype: {}.", mimeType);
        }

        this.mimeType = mime;
    }

}
